### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by ganemone**

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -6,17 +6,37 @@
...
+        return new Promise(resolve => {
+          const oldEnd = ctx.res.end.bind(ctx.res);
+          ctx.res.end = (data, encoding, cb) => {
+            ctx.respond = false;
+            oldEnd(data, encoding, cb);
+            return next().then(resolve);
+          };
+          ctx.res.fusionRender = () => {
+            ctx.res.end = oldEnd;
+            next().then(resolve);
```
Comment (lhorie) [inline src/server.js]: Should this return?

Comment (ganemone) [reply src/server.js]: It doesn't really matter what it returns, since it will be called in express or http handlers

Comment (lhorie) [reply src/server.js]: What happens if a downstream throws?

Comment (ganemone) [reply src/server.js]: Probably should add a test for that - I will put up a new PR

**Pull request merged**